server/application/application.go

```diff
@@ -1112,15 +1115,26 @@ func (s *Server) validateAndNormalizeApp(ctx context.Context, app *appv1.Applica
 			return err
 		}
 	}
-
 	// If source is Kustomize add build options
 	kustomizeSettings, err := s.settingsMgr.GetKustomizeSettings()
 	if err != nil {
 		return fmt.Errorf("error getting kustomize settings: %w", err)
 	}
-	kustomizeOptions, err := kustomizeSettings.GetOptions(app.Spec.Source)
-	if err != nil {
-		return fmt.Errorf("error getting kustomize options from settings: %w", err)
+	kustomizeOptions := make([]*appv1.KustomizeOptions, 0)
```

Comment:
in the [proposal](https://github.com/argoproj/argo-cd/blob/master/docs/proposals/multiple-sources-for-applications.md#add-new-sources-field-in-application-spec) we agreed that if both are present we would ignore the source

Reply:
Sure, and that helps when someone directly `kubectl apply`s an Application manifest. But I think it would be helpful to the users if we threw an error if they tried to add both via the UI/CLI.

Reply:
As discussed offline, added error message for `argocd app sync` command.